pkg/controller/endpoint/endpoints_controller_test.go

```diff
@@ -2279,9 +2280,10 @@ func TestMultipleServiceChanges(t *testing.T) {
 	controller.onServiceUpdate(svc)
 
 	// Ensure the work queue has been processed by looping for up to a second to prevent flakes.
-	wait.PollImmediate(50*time.Millisecond, 1*time.Second, func() (bool, error) {
-		return controller.queue.Len() == 0, nil
-	})
+	wait.PollUntilContextTimeout(context.Background(), 50*time.Millisecond, 1*time.Second, true,
```

Comment:
and reuse it here

Reply:
sorry, I don't find the ctx in TestMultipleServiceChanges() function.